src/dynamics/response.py

```diff
@@ -35,13 +35,15 @@ def respond(
                 "numbering the number of columns in 'features'"
             )
         changeable_features = unwrap_or(self.changeable_features, default=[])
+        epsilon_map = unwrap_or(self.epsilon, default={a: 1 for a in list(changeable_features)})
         new_features = np.copy(features)
 
         neg_item_mask = np.dot(action[None, :], features[...].T)[0] < 0
         assert not isinstance(changeable_features, slice)
         for changeable_feature in changeable_features:
+            weight = epsilon_map[changeable_feature]
             new_features[np.nonzero(neg_item_mask), changeable_feature] += (
-                self.epsilon / np.linalg.norm(action)
+                weight / np.linalg.norm(action)
```

Comment:
I'm not sure on this line

Reply:
I guess the question is how should epsilon for this value be interpreted: as an absolute, or relative distance

Reply:
Hmm, I'm not sure I understand. The way you did it here seems like the most straight-forward way to do it.

Reply:
😅 I mean that I'm not sure about dividing by the norm of the entire weight vector, and then I was mostly talking to myself about the merits of just deleting it

Reply:
ah, that part. yeah, one reason to delete it is that Perdomo et al. don't have it (I think?)

Reply:
Leaving it for now, but have put a TODO comment  here for discussion at a later date